libs/acn/RPTPDU.cpp

```diff
@@ -0,0 +1,132 @@
+}
+
+
+/*
+ * Size of the data portion
+ */
+unsigned int RPTPDU::DataSize() const {
+  return m_pdu ? m_pdu->Size() : 0;
+}
+
+
+/*
+ * Pack the header portion.
+ */
+bool RPTPDU::PackHeader(uint8_t *data, unsigned int *length) const {
+  unsigned int header_size = HeaderSize();
+
+  if (*length < header_size) {
+    OLA_WARN << "RPTPDU::PackHeader: buffer too small, got " << *length
+             << " required " << header_size;
+    *length = 0;
+    return false;
+  }
+
+  RPTHeader::rpt_pdu_header header;
+  m_header.SourceUID().Pack(header.source_uid, sizeof(header.source_uid));
+  header.source_endpoint = HostToNetwork(m_header.SourceEndpoint());
+  m_header.DestinationUID().Pack(header.destination_uid,
+                                 sizeof(header.destination_uid));
+  header.destination_endpoint = HostToNetwork(m_header.DestinationEndpoint());
+  header.sequence = HostToNetwork(m_header.Sequence());
+  *length = sizeof(RPTHeader::rpt_pdu_header);
+  memcpy(data, &header, *length);
+  return true;
+}
+
+
+/*
+ * Pack the data portion.
+ */
+bool RPTPDU::PackData(uint8_t *data, unsigned int *length) const {
+  if (m_pdu)
+    return m_pdu->Pack(data, length);
+  *length = 0;
+  return true;
+}
+
+
+/*
+ * Pack the header into a buffer.
+ */
+void RPTPDU::PackHeader(OutputStream *stream) const {
+  RPTHeader::rpt_pdu_header header;
+  m_header.SourceUID().Pack(header.source_uid, sizeof(header.source_uid));
+  header.source_endpoint = HostToNetwork(m_header.SourceEndpoint());
+  m_header.DestinationUID().Pack(header.destination_uid,
+                                 sizeof(header.destination_uid));
+  header.destination_endpoint = HostToNetwork(m_header.DestinationEndpoint());
+  header.sequence = HostToNetwork(m_header.Sequence());
+  stream->Write(reinterpret_cast<uint8_t*>(&header),
+                sizeof(RPTHeader::rpt_pdu_header));
+}
+
+
+/*
+ * Pack the data into a buffer
+ */
+void RPTPDU::PackData(OutputStream *stream) const {
+  if (m_pdu)
+    m_pdu->Write(stream);
+}
+
+
+void RPTPDU::PrependPDU(ola::io::IOStack *stack,
+                        uint32_t vector,
+                        const ola::rdm::UID &source_uid,
+                        uint16_t source_endpoint,
+                        const ola::rdm::UID &destination_uid,
+                        uint16_t destination_endpoint,
+                        uint32_t sequence_number) {
+  RPTHeader::rpt_pdu_header header;
+  source_uid.Pack(header.source_uid, sizeof(header.source_uid));
+  header.source_endpoint = HostToNetwork(source_endpoint);
+  destination_uid.Pack(header.destination_uid, sizeof(header.destination_uid));
+  header.destination_endpoint = HostToNetwork(destination_endpoint);
+  header.sequence = HostToNetwork(sequence_number);
+  stack->Write(reinterpret_cast<uint8_t*>(&header),
+               sizeof(RPTHeader::rpt_pdu_header));
+
+  vector = HostToNetwork(vector);
+  stack->Write(reinterpret_cast<uint8_t*>(&vector), sizeof(vector));
```

Comment:
null-dereference? I seem to be quite sensitive to them today :D

Reply:
I've now trapped this one directly above.